### PR TITLE
Feature/aw/adding icon buttons

### PR DESCRIPTION
**Pull request by 6Botter7**
Added buttons with icons

### CODE REVIEW

File: styles/lib/buttons.css [hunk 1]
```diff
@@ -6,11 +16,18 @@ button {
     border-radius: var(--radius);
     padding: var(--padding);
     box-sizing: border-box;
-    height: 2.5rem;
+    height: 40px;
```
Comment (caperaven) [inline styles/lib/buttons.css]: use rem not px

File: styles/lib/buttons.css [hunk 2]
```diff
@@ -87,10 +104,7 @@ button.icon {
     font-family: crsfrw;
     display: flex;
     align-items: center;
-    justify-content: center;
-    background-color: var(--alt-fg);
-    border: 0;
-    font-size: 1rem;
+    font-size: 20px;
```
Comment (caperaven) [inline styles/lib/buttons.css]: why is this px?

File: styles/lib/buttons.css [hunk 3]
```diff
@@ -100,4 +114,48 @@ button.icon svg {
 
 button.transparent {
     background: transparent;
+}
+
+
+button[data-icon-left]::before, button[data-icon-right]::after {
+    font-family: crsfrw;
+    font-size: 20px;
```
Comment (caperaven) [inline styles/lib/buttons.css]: why is this px?

File: styles/lib/buttons.css [hunk 4]
```diff
@@ -100,4 +114,48 @@ button.icon svg {
...
+
+button[data-icon-right="loader"]::after {
+    animation: busy 0.5s linear infinite;
+}
+
+button.error::after {
+    content: "info-outline";
+    margin-left: 0.5rem;
+    font-family: crsfrw;
+    font-size: 20px;
```
Comment (caperaven) [inline styles/lib/buttons.css]: px?

File: styles/lib/buttons.css [hunk 5]
```diff
@@ -100,4 +114,48 @@ button.icon svg {
...
+button.success::after {
+    content: "check";
+    margin-left: 0.5rem;
+    font-family: crsfrw;
+    font-size: 20px;
+}
+
+@keyframes busy {
+    0% {
+        transform: rotate(0deg);
```
Comment (caperaven) [inline styles/lib/buttons.css]: don't use transform here just use rotate directly instead please

**Review by caperaven — DISMISSED** [review]

File: styles/lib/buttons.css [hunk 6]
```diff
@@ -8,9 +18,16 @@ button {
     box-sizing: border-box;
     height: 2.5rem;
     min-width: 2rem;
+    margin: 0 5px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
     font-weight: bold;
+    font-size: 1rem;
+    font-family: 'SourceSansPro', sans-serif;
```
Comment (Ashton-Pragma) [inline styles/lib/buttons.css]: do we need this?

**Review by Ashton-Pragma — APPROVED** [review]

**Review by Kieran-Reilly — APPROVED** [review]

**Pull request merged**